testing/tests/DevExpress.ui.widgets/multiView.tests.js

```diff
@@ -749,6 +773,71 @@ QUnit.module('loop', {
         assert.equal(multiView.option('selectedIndex'), 1, 'selected index changed correctly');
     });
 
+    QUnit.test('selected index should be set to the first item after right swipe from the second if the last is disabled', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [1, 2, { disabled: true, text: 3 }],
+            selectedIndex: 1,
+            loop: true,
+        });
+        const multiView = $('#multiView').dxMultiView('instance');
+        const pointer = pointerMock($multiView);
+
+        pointer.start().swipeStart().swipe(0.5).swipeEnd(1);
+        assert.strictEqual(multiView.option('selectedIndex'), 0, 'selected index changed correctly');
+    });
+
+    QUnit.test('selected index should be set to the second item after left swipe from the first', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [1, 2, 3, 4, 5],
+            selectedIndex: 0,
+            loop: true,
+        });
+        const multiView = $('#multiView').dxMultiView('instance');
+        const pointer = pointerMock($multiView);
+
+        pointer.start().swipeStart().swipe(-0.5).swipeEnd(-1);
+        assert.strictEqual(multiView.option('selectedIndex'), 1, 'selected index changed correctly');
+    });
+
+    QUnit.test('selected index should be set to the fourth item after right swipe from the last', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [1, 2, 3, 4, 5],
+            selectedIndex: 4,
+            loop: true,
+        });
+        const multiView = $('#multiView').dxMultiView('instance');
+        const pointer = pointerMock($multiView);
+
+        pointer.start().swipeStart().swipe(0.5).swipeEnd(1);
+        assert.strictEqual(multiView.option('selectedIndex'), 3, 'selected index changed correctly');
+    });
+
+    QUnit.test('selected index should be set to the first item after left swipe from the last, if the last is disabled', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [1, 2, 3, 4, { disabled: true }],
+            selectedIndex: 4,
+            loop: true,
+        });
+        const multiView = $('#multiView').dxMultiView('instance');
+        const pointer = pointerMock($multiView);
+
+        pointer.start().swipeStart().swipe(-0.5).swipeEnd(-1);
+        assert.strictEqual(multiView.option('selectedIndex'), 0, 'selected index changed correctly');
+    });
+
+    QUnit.test('selected index should be set to the fifth item after right swipe from the first, if the first is disabled', function(assert) {
+        const $multiView = $('#multiView').dxMultiView({
+            items: [{ disabled: true }, 2, 3, 4, 5],
```

Comment:
Is it real case that disabled item can be selected?

Reply:
Yes